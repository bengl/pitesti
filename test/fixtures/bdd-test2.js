'use strict';

/* eslint-disable */

const assert = require('assert');
const {
  before, beforeAll, beforeEach,
  after, afterAll, afterEach,
  describe, context, it
} = require('../../bdd');

const expectedStack = [
  -1,
  'before0',
  'beforeAll0',
  'describe1before',
  'beforeEach0',
  'describe1beforeEach',
  'describe1ctx2test1',
  'describe1afterEach',
  'beforeEach0',
  'describe3test1',
  'after0',
  'afterAll0'
];

const stack = [];
function pushStack (n) {
  stack.push(n);
}
pushStack(-1);
before(() => pushStack('before0'));
beforeAll(() => pushStack('beforeAll0'));
beforeEach(() => pushStack('beforeEach0'));
after(() => pushStack('after0'));
afterAll(() => {
  pushStack('afterAll0');
  assert.deepStrictEqual(expectedStack, stack);
});
describe('describe1', () => {
  before(() => pushStack('describe1before'));
  beforeEach(() => pushStack('describe1beforeEach'));
  afterEach(() => pushStack('describe1afterEach'));
  context('ctx1', () => {
    it('test1', () => pushStack('describe1ctx1test1'));
  });
  describe('ctx2', () => {
    it.only('test1', () => pushStack('describe1ctx2test1'));
  });
});
describe('describe2', () => {
  before(() => pushStack('describe2before'));
  beforeEach(() => pushStack('describe2beforeEach'));
  afterEach(() => pushStack('describe2afterEach'));
  context('ctx1', () => {
    it('test1', () => pushStack('describe2ctx1test1'));
  });
  it('test1', () => pushStack('describe2test1'));
});
describe.only('describe3', () => {
  it('test1', () => pushStack('describe3test1'));
});
