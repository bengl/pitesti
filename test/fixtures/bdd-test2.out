TAP version 13
1..24
ok 1 [before]
ok 2 [before]
ok 3 describe1 [before]
ok 4 describe1 ctx1 test1 [beforeEach] # SKIP
ok 5 describe1 ctx1 test1 [beforeEach] # SKIP
ok 6 describe1 ctx1 test1 # SKIP
ok 7 describe1 ctx1 test1 [afterEach] # SKIP
ok 8 describe1 ctx2 test1 [beforeEach]
ok 9 describe1 ctx2 test1 [beforeEach]
ok 10 describe1 ctx2 test1
ok 11 describe1 ctx2 test1 [afterEach]
ok 12 describe2 [before] # SKIP
ok 13 describe2 ctx1 test1 [beforeEach] # SKIP
ok 14 describe2 ctx1 test1 [beforeEach] # SKIP
ok 15 describe2 ctx1 test1 # SKIP
ok 16 describe2 ctx1 test1 [afterEach] # SKIP
ok 17 describe2 test1 [beforeEach] # SKIP
ok 18 describe2 test1 [beforeEach] # SKIP
ok 19 describe2 test1 # SKIP
ok 20 describe2 test1 [afterEach] # SKIP
ok 21 describe3 test1 [beforeEach]
ok 22 describe3 test1
ok 23 [after]
ok 24 [after]

# tests 24
# pass  11
# fail  0
# skip  13
